### PR TITLE
Feature/safe mode

### PR DESCRIPTION
**Pull request by koallann**
To enable safe mode:

- Don't call `Builder.setToken()`
- `Builder.setSafeModeEnabled(true)`

### CODE REVIEW

File: fcm-channel/src/main/java/io/fcmchannel/sdk/chat/FcmClientChatFragment.java [hunk 1]
```diff
@@ -119,8 +119,10 @@ private void setupView(View view) {
 
         message = view.findViewById(R.id.message);
         adapter = new ChatMessagesAdapter(chatUiConfiguration, onMetadataItemClickListener);
-        adapter.showLoading();
 
+        if (!FcmClient.isSafeModeEnabled()) {
+            adapter.showLoading();
```
Comment (koallann) [inline fcm-channel/src/main/java/io/fcmchannel/sdk/chat/FcmClientChatFragment.java:124]: Display messages list loading when we can request messages from endpoint.

File: fcm-channel/src/main/java/io/fcmchannel/sdk/FcmClient.java [hunk 2]
```diff
@@ -252,6 +269,10 @@ public static boolean hasFloatingPermission() {
             || (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && Settings.canDrawOverlays(context));
     }
 
+    public static String getFcmToken() {
```
Comment (koallann) [inline fcm-channel/src/main/java/io/fcmchannel/sdk/FcmClient.java:272]: This method was added because FCM token was removed from shared preferences.

Comment (johncordeiro) [reply fcm-channel/src/main/java/io/fcmchannel/sdk/FcmClient.java:272]: well done

File: fcm-channel/src/main/java/io/fcmchannel/sdk/services/FcmClientRegistrationIntentService.java [hunk 3]
```diff
@@ -38,22 +38,22 @@ protected void onHandleIntent(Intent intent) {
...
 
             Preferences preferences = FcmClient.getPreferences();
             Contact contact = null;
 
-            if (!TextUtils.isEmpty(FcmClient.getToken())) {
-                Response<FcmRegistrationResponse> response = saveContactWithToken(urn, fcmToken, contactUuid);
+            Response<FcmRegistrationResponse> response = saveContactWithToken(urn, fcmToken, contactUuid);
+
+            if (response.isSuccessful()) {
+                contact = new Contact();
```
Comment (johncordeiro) [inline fcm-channel/src/main/java/io/fcmchannel/sdk/services/FcmClientRegistrationIntentService.java:49]: where this contact object is used?

Comment (koallann) [reply fcm-channel/src/main/java/io/fcmchannel/sdk/services/FcmClientRegistrationIntentService.java:49]: Here: https://github.com/push-platform/fcm-channel-android/pull/35/files#diff-a2526e0d221caae34088316e4a04b744R81

That way the lib client can extends this service and override onFcmRegistered to perform some operation, like: https://github.com/push-platform/fcm-channel-android/blob/master/sample/src/main/java/io/fcmchannel/sdk/sample/services/PushRegistrationService.java#L18

**Review by johncordeiro — APPROVED** [review]

**Pull request merged**